### PR TITLE
Bump htmx to 2.0.0

### PR DESCRIPTION
**Pull request by yardenshoham**
Tested Subscribe, Follow, Star, Watch, and System Status.

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Comment by silverwind** [comment]
Does it work in PaleMoon?

**Comment by silverwind** [comment]
Also docs need a small replacement from `hx-on` to `hx-on:` here as per [this](https://htmx.org/migration-guide-htmx-1/):

```
docs/content/contributing/guidelines-frontend.zh-cn.md:75:* htmx + 任何其他需要大量 JavaScript 代码或不必要的功能，如 htmx 脚本 (`hx-on`)
docs/content/contributing/guidelines-frontend.en-us.md:75:* htmx + any other framework which requires heavy JS code, or unnecessary features like htmx scripting (`hx-on`)
```

**Review by a1012112796 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by yardenshoham** [comment]
> Also docs need a small replacement from `hx-on` to `hx-on:` here as per [this](https://htmx.org/migration-guide-htmx-1/):
> 
> ```
> docs/content/contributing/guidelines-frontend.zh-cn.md:75:* htmx + 任何其他需要大量 JavaScript 代码或不必要的功能，如 htmx 脚本 (`hx-on`)
> docs/content/contributing/guidelines-frontend.en-us.md:75:* htmx + any other framework which requires heavy JS code, or unnecessary features like htmx scripting (`hx-on`)
> ```

It's OK, we forbid all use of `hx-on`, doesn't matter which syntax of it

**Comment by silverwind** [comment]
Will test in PaleMoon

**Comment by silverwind** [comment]
I see two errors when testing this branch. Firefox:

image

PaleMoon (bug will be fixed next version of PaleMoon):

image

**Comment by silverwind** [comment]
The first error is related to idiomorph extension:

image

I doubt this was actually tested...

**Review by silverwind — CHANGES_REQUESTED** [review]
Blocking because of unresolved bug with idiomorph extension.

**Comment by yardenshoham** [comment]
I tested only in chrome

**Comment by silverwind** [comment]
> The first error is related to idiomorph extension:
> image
> 
> I doubt this was actually tested...

https://github.com/go-gitea/gitea/pull/31413#issuecomment-2178713865 shows in Chrome as well for me.

**Comment by yp05327** [comment]
I can also get this error in Chrome (windows)

**Comment by yp05327** [comment]
It seems that this is caused by https://github.com/bigskysoftware/htmx/issues/2629

**Comment by silverwind** [comment]
Missing typescript definitions can not cause such problems. To me it seems https://github.com/bigskysoftware/idiomorph is simply not compatible with htmx 2.x yet.

**Comment by silverwind** [comment]
`defineExtension` [still exists](https://github.com/bigskysoftware/htmx/blob/fa2dc6c02e96786d7daaae7c165daef229829250/src/htmx.js#L4809) in the source, but I think it's not exposed as `htmx.defineExtension`, so idk. I'm quite fed up with how bad htmx docs is are and how amateurish that project is being managed.

**Comment by silverwind** [comment]
Fix pushed in https://github.com/go-gitea/gitea/pull/31413/commits/e32f872690cf353dad0eab40588cfdb292ca4a11. The problem apparently is that htmx no longer sets the `htmx` global, at least not in our setup. I changed it so the global is now set explicitely and in the same manner as we already do for jQuery.

**Review by silverwind — APPROVED** [review]

File: web_src/js/globals.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import $ from 'jquery';
+import htmx from 'htmx.org/dist/htmx.esm.js';
+
+window.$ = window.jQuery = $; // eslint-disable-line no-jquery/variable-pattern
+window.htmx = htmx;
```
Comment (wxiaoguang) [inline web_src/js/globals.js]:
```suggestion
import jquery from 'jquery';
import htmx from 'htmx.org/dist/htmx.esm.js';

window.$ = window.jQuery = jquery;
window.htmx = htmx;
```

And the old `jquery.js` should be removed?

File: webpack.config.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ export default {
   mode: isProduction ? 'production' : 'development',
   entry: {
     index: [
-      fileURLToPath(new URL('web_src/js/jquery.js', import.meta.url)),
+      fileURLToPath(new URL('web_src/js/globals.js', import.meta.url)),
```
Comment (wxiaoguang) [inline webpack.config.js:79]: Here, the old jquery.js should be removed?

**Review by delvh — APPROVED** [review]

File: webpack.config.js (hunk 2 above)
Comment (silverwind) [reply webpack.config.js:79]: yes indeed, i forgot to `git rm` it.

**Pull request merged**